docs/src/main/docs/health/01_health.adoc

```diff
@@ -20,71 +20,192 @@
+ health check and returns a response with a status code representing the overall
+ state of the application.
+
+[cols="1,5",role="flex, sm7"]
+.Health status codes
+|=======
+| `200` | All health checks succeeded
+| `503` | At least one health check failed
+| `500` | An error occurred while reporting the health
+|=======
 
-=== Adding health checks to your application
-To enable health checks, register them with the web server.
+The HTTP response also contains a JSON payload that describes the statuses for
+ all health checks.
 
 [source,java]
+.Create the health support service
 ----
-import io.helidon.metrics.MetricsSupport;
-//...
-HealthSupport health = HealthSuport.builder()
-    .add(HealthChecks.healthChecks())               // <1>
+HealthSupport health = HealthSupport.builder()
+    .add(this::exampleHealthCheck)
+    .build();
+----
+
+NOTE: Health check requires the `JSON-P` support to be enabled, see the example
+ below.
+
+=== Example
+
+[source,java]
+.Register a custom health check
+----
+HealthSupport health = HealthSupport.builder()
     .add(() -> HealthCheckResponse.named("exampleHealthCheck")
                  .up()
                  .withData("time", System.currentTimeMillis())
-                 .build())                          // <2>
+                 .build()) // <1>
     .build();
 
 Routing.builder()
-        .register(health)                           // <3>
-        .register("/hello", (req, res) -> res.send("HelloWorld"))
+        .register(JsonSupport.create()) // <2>
+        .register(health) // <3>
         .build();
 ----
-<1> Add built-in health checks (requires the `helidon-health-checks` dependency)
-<2> Add a custom health check. This example returns `UP` and current time
-<3> Register health support with web server routing (adds the `/health` endpoint)
+<1> Add a custom health check. This example returns `UP` and current time
+<2> Enable support for `JSON`
+<3> Register health support with web server routing (adds the `/health`
+ endpoint)
 
-=== Accessing the health check endpoint
+TIP: Remember to balance the amount of collected information in order to avoid
+ overloading the application and impacting users.
 
-You can get health-check data through the `/health` endpoint.
+[source,json]
+.JSON response
+----
+{
+    "outcome": "UP",
+    "checks": [
+        {
+            "name": "exampleHealthCheck",
+            "state": "UP",
+            "data": {
+                "time": 1546958376613
+            }
+        }
+    ]
+}
+----
+
+=== Built-in health-checks
+
+A set of built-ins health checks can be optionally enabled to report various
+ health check statuses that are commonly used:
+
+* Deadlock detection
+* Available disk space
+* Available heap memory
 
-The following is an example of the response (in JSON format) to a health-check request, based on the built-in and custom
-    health checks implemented in the example provided earlier:
+[source,java]
+.Register built-in health-checks
```

Comment:
Earlier (line 48) is a source block without a title because the purpose is clear from the context, followed by another source block _with_ a title. Here, even the first one has a title that doesn't add much given the preceding text.

We should be consistent if we can. Always omit the title if the purpose is clear from the context, or always use a title for every source block, but we shouldn't vary if we can avoid it. There is less clutter without a redundant title (as long as the purpose is indeed clear from the context.)

Reply:
Ok, I will remove the title.